### ISSUE TITLE
New rule: prevent unsafe assignment of fp-ts types

### ISSUE THREAD

**Issue by danielnixon** [danielnixon/eslint-plugin-total-functions#728]
```ts
export const foo: IO<string> = () => "hello";

// These assignments compile because the types are compatible, but the semantics are such that doing so is almost certainly an error.
export const bar: Lazy<string> = foo;
export const baz: () => string = foo;
```

**Comment by danielnixon** [comment]
We should also catch when a `Promise` is being assigned to an IO like `IO.of(Axios.get(...))`, or any async effect being assigned to an IO for that matter.

**Comment by danielnixon** [comment]
That latter point is covered by `total-functions/no-nested-fp-ts-effects`
